addons/mass_mailing/models/mail_mail.py

```diff
@@ -95,11 +95,13 @@ def _send_prepare_values(self, partner=None):
         return res
 
     @api.multi
-    def _postprocess_sent_message(self, mail_sent=True):
+    def _postprocess_sent_message(self, failure_type='NONE', **kwargs):
+        mail_sent = failure_type in ('NONE', 'RECIPIENT')  # TODO xdo initial behaviour to check (need mail sent? )
```

Comment:
The idea here was that the mail_sent was true if at least one recipient received it, even if one of the recipient was invalid (or all exept one). This is not perfectly the case but using failure_type in ('NONE', 'RECIPIENT') follows almost  the previous logic ( only use case that will change the logic: all recipient are invalid)

Agree with you for the second part but I'm not really aware of for what statistics_ids are used for, need to investigate first (or precise specs).

Reply:
statistics_ids are just traces generated by mass mailing. Each sent email creates a statistics that stores the read/ sent / bounce / exception date. It is used to compute mass mailing statistics: number of people who opened the email, read it, ... it is also used to trigger the activities in marketing automation.

It is a bit like notifications, but linked to email instead of partners. A next step could be to see if we don't have duplicated info. I am not sure we should merge them as those are used for separate features. Anyway that's not the purpose of this task :) .

Concerning the mail_sent: as mass mailing emails are sent to one recipient, I would say that recipient is now an error throwing the statistics in exception. I prefer people having too much errors on mass mailing than having "everything is ok" whereas no email was sent.